src/Development/IDE/Core/Compile.hs

```diff
@@ -95,8 +99,17 @@ typecheckModule (IdeDefer defer) packageState deps pm =
     runGhcEnv packageState $
         catchSrcErrors "typecheck" $ do
             setupEnv deps
+#if MIN_GHC_API_VERSION(8,6,0)
```

Comment:
Any way to abstract the key steps out into a function with a single signature that is the same on all versions (merely a noop on one version), then have this code just call that? Would make managing the CPP a bit easier.